### PR TITLE
roachprod: run scheduled backup init without timeout

### LINKED ISSUE #97010: roachtest: tpccbench/nodes=12/cpu=16 failed
roachtest.tpccbench/nodes=12/cpu=16 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestWeeklyBazel/8668054?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestWeeklyBazel/8668054?buildTab=artifacts#/tpccbench/nodes=12/cpu=16) on master @ [2a7edbeb0737b1309064c25c641a309c2980d9ba](https://github.com/cockroachdb/cockroach/commits/2a7edbeb0737b1309064c25c641a309c2980d9ba):

```
test artifacts and logs in: /artifacts/tpccbench/nodes=12/cpu=16/run_1
(cluster.go:1864).Start: parallel execution failure: ~ ./cockroach sql --url 'postgres://root@localhost:26257?sslmode=disable' "-e
CREATE SCHEDULE IF NOT EXISTS test_only_backup FOR BACKUP INTO 'gs://cockroachdb-backup-testing/roachprod-scheduled-backups/teamcity-8668054-1676096165-11-n13cpu16/1676158192593320348?AUTH=implicit' RECURRING '*/15 * * * *'
FULL BACKUP '@hourly'
WITH SCHEDULE OPTIONS first_run = 'now'"
ERROR: cannot dial server.
Is the server running?
If the server is running, check --host client-side and --advertise server-side.
dial tcp 127.0.0.1:26257: connect: connection refused
Failed running "sql": COMMAND_PROBLEM: ssh verbose log retained in ssh_232952.593470062_n1_run-sql.log: exit status 1
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=16</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_fs=ext4</code>
, <code>ROACHTEST_localSSD=true</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*tpccbench/nodes=12/cpu=16.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-24460

### PR DESCRIPTION
**Pull request by msbutler**
Previously, several roachtests failed during a cluster restart because a node serving the default scheduled backup command was not ready to serve requests. At this time, when roachprod start returns, not every node may be ready to serve requests.

To prevent this failure mode, this patch changes the scheduled backup cmd during roachprod.Start() to run with infinite timeout and only on the the first node in the cluster.

Fixes #97010, #97232

Release note: None

Epic: none

### CODE REVIEW

**Comment by msbutler** [comment]
@renatolabs this patch seems to work on a few roachtests which run restarts. Happy run it on the whole nightly suite if you'd like.

File: pkg/roachprod/install/cockroach.go [hunk 1]
```diff
@@ -817,24 +814,28 @@ func (c *SyncedCluster) createFixedBackupSchedule(
...
+
+	node := Node(1)
+	binary := cockroachNodeBinary(c, node)
+	url := c.NodeURL("localhost", c.NodePort(node), "" /* tenantName */)
+	fullCmd := fmt.Sprintf(`COCKROACH_CONNECT_TIMEOUT=0 %s sql --url %s -e %q`,
+		binary, url, createScheduleCmd)
+	// Instead of using `c.ExecSQL()`, use the more flexible c.Run(), which allows us to
+	// 1) prefix the schedule backup cmd with COCKROACH_CONNECT_TIMEOUT=0.
+	// 2) run the command against the first node on the cluster.
+	l.Printf("%s\n", fullCmd)
```
Comment (renatolabs) [inline pkg/roachprod/install/cockroach.go:837]: I don't think you intended to commit this?

**Review by renatolabs — APPROVED** [review]
LGTM!

File: pkg/roachprod/install/cockroach.go (hunk 1 above)
Comment (msbutler) [reply pkg/roachprod/install/cockroach.go]: oops. resolved.

**Comment by msbutler** [comment]
TFTR!

bors r=renatolabs

**Pull request merged**